packages/components/src/test/viewport/WithUnifiedSelection.test.tsx

```diff
@@ -5,161 +5,184 @@
-    imodelMock.setup((imodel) => imodel.views).returns(() => viewsMock.object);
+  beforeEach(() => {
+    views.load.resolves({} as ViewState3d);
   });
 
   afterEach(() => {
+    views.load.reset();
     sinon.restore();
   });
 
   it("renders", () => {
-    render(<PresentationViewport imodel={imodelMock.object} viewDefinitionId={viewDefinitionId} selectionHandler={selectionHandlerMock.object} />);
+    render(<PresentationViewport imodel={imodel} viewDefinitionId={viewDefinitionId} selectionHandler={selectionHandler} />);
   });
 
   it("creates default ViewportSelectionHandler implementation when not provided through props", () => {
-    const selectionManagerMock = moq.Mock.ofType<SelectionManager>();
     const selectionChangeEvent = new SelectionChangeEvent();
-    selectionManagerMock.setup((x) => x.selectionChange).returns(() => selectionChangeEvent);
-    selectionManagerMock.setup((x) => x.suspendIModelToolSelectionSync(imodelMock.object)).returns(() => ({ dispose: () => {} }));
-    selectionManagerMock.setup(async (x) => x.getHiliteSet(imodelMock.object)).returns(async () => ({}));
-    sinon.stub(Presentation, "selection").get(() => selectionManagerMock.object);
+    const selectionManagerMock = {
+      selectionChange: selectionChangeEvent,
+      suspendIModelToolSelectionSync: () => ({ dispose: () => {} }),
+      *getHiliteSetIterator() {},
+      setSyncWithIModelToolSelection: () => {},
+    };
+    sinon.stub(Presentation, "selection").get(() => selectionManagerMock);
 
     expect(selectionChangeEvent.numberOfListeners).to.be.eq(0);
 
-    render(<PresentationViewport imodel={imodelMock.object} viewDefinitionId={viewDefinitionId} />);
+    render(<PresentationViewport imodel={imodel} viewDefinitionId={viewDefinitionId} />);
 
     // new 'ViewportSelectionHandler' should be listening to selection change event
     expect(selectionChangeEvent.numberOfListeners).to.be.eq(1);
   });
 
   it("sets ViewportSelectionHandler.imodel property when rendered with new imodel", () => {
-    const { rerender } = render(
-      <PresentationViewport imodel={imodelMock.object} viewDefinitionId={viewDefinitionId} selectionHandler={selectionHandlerMock.object} />,
-    );
-    selectionHandlerMock.verify((x) => (x.imodel = imodelMock.object), moq.Times.atLeastOnce());
-
-    const newImodel = moq.Mock.ofType<IModelConnection>();
-    rerender(<PresentationViewport imodel={newImodel.object} viewDefinitionId={viewDefinitionId} selectionHandler={selectionHandlerMock.object} />);
+    const { rerender } = render(<PresentationViewport imodel={imodel} viewDefinitionId={viewDefinitionId} selectionHandler={selectionHandler} />);
+    expect(selectionHandler.imodel).to.be.eq(imodel);
 
-    selectionHandlerMock.verify((x) => (x.imodel = newImodel.object), moq.Times.atLeastOnce());
-  });
-
-  it("disposes ViewportSelectionHandler when unmounted", () => {
```

Comment:
do you think this is not worth checking?

Reply:
I don't think `viewWithUnifiedSelection` should disposed `ViewportSelectionHandler` received through props because that won't work in React 18 `StrictMode`

Reply:
Agree. I guess there's no way to check if we dispose the handler in cases we create it ourselves?

Reply:
Updated test to check that default handler is disposed [003ef54](https://github.com/iTwin/presentation/pull/354/commits/003ef54dd5ea40d772e61e39421d01195876aec5)